lib/upgradedb/mysql.2021050701.php

```diff
@@ -23,13 +23,19 @@
 
 $this->BeginTrans();
 
+if ($this->ResourceExists(
+    'customerassignment',
+    LMSDB::RESOURCE_TYPE_CONSTRAINT)
+{
+    $this->Execute("ALTER TABLE customerassignments DROP CONSTRAINT customerassignment");
+}
+
 if (!$this->ResourceExists('vcustomerassignments', LMSDB::RESOURCE_TYPE_VIEW)) {
```

Comment:
No dobra - tu wykrywasz czy istnieje pod tą nazwą. Patrzyłeś czy nie istnieje jeszcze czasem pod inną nazwą?

Reply:
Teraz ok?

Bez commita 156f8f508baa9fcf657f254bacd2ff8811642b59 mam błąd:
```
upgradedb.php
(C) 2001-2022 LMS Developers
Using file /etc/lms/lms.ini as config.
DB schema version bumped to 2022062807
DB schema upgrade errors:
ALTER TABLE voip_cdr DROP CONSTRAINT voip_cdr_uniqueid_key
ERROR:  constraint "voip_cdr_uniqueid_key" of relation "voip_cdr" does not exist
ALTER TABLE voip_cdr ADD CONSTRAINT voip_cdr_type_uniqueid_ukey UNIQUE (type, uniqueid)
ERROR:  current transaction is aborted, commands ignored until end of transaction block
UPDATE dbinfo SET keyvalue = '2022062808' WHERE keytype = 'dbversion'
ERROR:  current transaction is aborted, commands ignored until end of transaction block
```

Reply:
Nie szukaj w źródłach tylko w aktualnej wersji schematu bazy danych tuż przed nieszczęsną aktualizacją schematu. Może nazwa constrainta jest inna. Nie odnalezienie go i tym samym nieusunięcie może spowodować w przyszłości różne dziwne problemy.

Reply:
Przypadek 1:
`git reset --hard 'origin/master@{2022-06-28 00:00:00}'`

W mysql.2022062808.php sprawdzamy czy jest voip_cdr_type_uniqueid_ukey (a powinniśmy sprawdzać czy jest uniqueid) bo jego dropujemy.

W postgresql.2022062808.php sprawdzamy czy jest voip_cdr_type_uniqueid_ukey (a powinniśmy sprawdzać czy jest voip_cdr_uniqueid_key) bo jego dropujemy.

Przypadek 2:
`git reset --hard 'origin/master@{2021-05-07 00:00:00}'`

W mysql.2021050701.php nie sprawdzaliśmy czy jest constraint customerassignment tylko go od razu dropowaliśmy co powodowało błąd bo ten constraint nie był nigdy tworzony.

W postgres.2021050701.php nie sprawdzaliśmy czy istnieje constraint customerassignments_customergroupid_key tylko go od razu dropowaliśmy.

@chilek oba naprawione są tutaj.

Reply:
Przesiedziałem nad tym kolejny kwadrans. Nie widzę nic błędnego w tym PR.

Reply:
@chilek wypuścisz to lub pokażesz jak być powinno? to jest mój jedyny bloker do update

Reply:
Jeszcze raz: skoro na tym etapie nie istnieje u Ciebie CONSTRAINT `customerassignment` to może to oznaczać, że istnieje pod inną nazwą. Sprawdzałeś przed tą aktualizacją schematu DB czy nie istnieje u Ciebie ten CONSTRAINT pod inną nazwą?

Reply:
> Przesiedziałem nad tym kolejny kwadrans. Nie widzę nic błędnego w tym PR.

Nawet jak byś 10h przesiedział to nic by to nie dało, bo chyba zajmujesz się czymś innym niż ja wspomniałem :)

Reply:
> Jeszcze raz: skoro na tym etapie nie istnieje u Ciebie CONSTRAINT `customerassignment` to może to oznaczać, że istnieje pod inną nazwą. Sprawdzałeś przed tą aktualizacją schematu DB czy nie istnieje u Ciebie ten CONSTRAINT pod inną nazwą?

Tak - opisałem to komentarzu wyżej w: https://github.com/chilek/lms/pull/2319/files#r976392905
 
> Jeśli nie usuniesz CONSTRAINT-a pod inną nazwą to całkiem prawdopodobnie, że w przyszłości doprowadzi do problemów.

Ale nie ma innej nazwy???